### ISSUE TITLE
Manual installation of pyproject-metadata dependency

### ISSUE THREAD

**Issue by rebecca-burwei** [scikit-build/scikit-build-core#427]
I'm not sure what the expectation is, but I had to install the pyproject-metadata dependency manually from: https://pypi.org/project/pyproject-metadata/

**Comment by rebecca-burwei** [comment]
Duplicate of https://github.com/scikit-build/scikit-build-core/issues/280

**Comment by rebecca-burwei** [comment]
Providing some more info about how I got this error. I followed the instructions for `C` [here](https://scikit-build-core.readthedocs.io/en/latest/getting_started.html#), and on the last step, got:

```{bash}
python3 -m pip install .
Processing /Users/rburwei/repos/my-c-project
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      Error in sitecustomize; set PYTHONVERBOSE for traceback:
      AssertionError:
      Traceback (most recent call last):
        File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 164, in prepare_metadata_for_build_wheel
          return hook(metadata_directory, config_settings)
        File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/scikit_build_core/build/__init__.py", line 71, in prepare_metadata_for_build_wheel
          from .wheel import _build_wheel_impl
        File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/scikit_build_core/build/wheel.py", line 22, in <module>
          from ..settings.metadata import get_standard_metadata
        File "/usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/scikit_build_core/settings/metadata.py", line 5, in <module>
          from pyproject_metadata import StandardMetadata
      ModuleNotFoundError: No module named 'pyproject_metadata'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip available: 22.2.2 -> 23.2
[notice] To update, run: python3.10 -m pip install --upgrade pip
```

My pyproject.toml file looks like:
```
[build-system]
requires = ["scikit-build-core"]
build-backend = "scikit_build_core.build"

[project]
name = "example"
version = "0.0.1"
```

Running `pip install pyproject-metadata` makes the error go away.

**Comment by rebecca-burwei** [comment]
We fixed this by updating homebrew.
